hc_core/src/error.rs

```diff
@@ -0,0 +1,62 @@
+use std::error::Error;
+use std::fmt;
+
+/// starter module for holding Holochain specific errors
+
+#[derive(Debug, PartialEq)]
+pub struct HolochainError {
+    details: String,
+}
```

Comment:
I've seen errors handled pretty well as enums in rust, which lets us provide composite results, and make better use of the trailing `?` upgrade operator... so, something like:

```rust
pub enum HolochainError {
  ErrorGeneric(String),
  SerdeError(serde_json::error::Error),
}

impl From<serde_json::error::Error> for HolochainError {
  fn from(err: serde_json::error::Error) -> Self {
    HolochainError::SerdeError(err)
  }
}
```

and then our own function like:

```rust
pub fn my_func() -> Result<(), HolochainError> {
  // call to fn that returns serde_json::error::Error can be upgraded to return holochain error
  let result = serde_json::to_string(something)?;
  // if error, we returned already, so we don't have to unwrap or anything
  // now we can just use result
}
```

Reply:
great idea, I'll change it.

Reply:
ok @neonphog just pushed a commit with Enum based implementation.

Reply:
nice, looks good :+1: